### DISCUSSION TITLE
Import all activation functions from linen.activations in linen.__init__

### DISCUSSION THREAD

**Discussion by onnoeberhard** [google/flax#2087] [Q&A] [answered]
I was trying to import the `selu` activation function from `flax.linen`, but found that it was not in the namespace. After inspection, it looks like `flax.linen.__init__` imports most, but not all, activation functions from `flax.linen.activations` (where they are in turn imported from `jax.nn`).

Is there a reason for why not all activation functions are imported? It seems odd to me that, when importing `flax.linen as nn`, I can get some like `nn.relu` but for others I have to use `jax.nn.selu`, especially as almost all of these activation functions are just imported from `jax.nn`.

If this is not intentional, I would be happy to do a quick pull request where all activation functions from `flax.linen.activations` are imported into the `flax.linen` namespace. If it is intentional, I would be interested in the reasoning of re-importing activation functions at all, and not just sticking with `jax.nn.relu`, for example.

**Comment by marcvanzee** [comment, marked as answer]
The reason why we re-import these functions (and others) from JAX rather than importing them directly is so we can potentially change the implementation if we want to, while remaining backward compatible. Moreover, we can also add our own custom functions in the same place (e.g., `PreLU`).

As for the shortcut in `__init__`, we do this for quite a number of functions actually. For instance, for initializers we do allow the shortcut `nn.ones`, but we don't allow it for `nn.initializers.kaiming_uniform`.

I agree it can be a bit confusing, but on the other hand, it does make sense that we do not always re-export everything in `__init__` to avoid poluting the namespace.

@levskaya @jheek please chime in if you think I missed anything (I wasn't around when this was implemented).

**Comment by onnoeberhard** [comment]
Oh I see, I had only been thinking about the activation functions and thought "if it's good enough for the `jax.nn` namespace, why not also `flax.linen`", but yeah if it also imports intializers and other functions, it might indeed get too crowded.  

By the way, [here](https://github.com/google/jax/blob/main/jax/nn/__init__.py) is the list of functions imported into `jax.nn`. It's mostly just a few more activation functions. I still think it might not be a bad idea to import these functions too, such that a user can simply "forget" about `jax.nn` (especially as it is encouraged to import `flax.linen as nn`), but this is a design decision with no obvious right answer.

**Reply by marcvanzee** [reply]
Sure, I am fine importing that list of activation functions into `__init__` for consistency with JAX. Do you want to file a PR?

**Reply by onnoeberhard** [reply]
Great, I created a pull request: #2088
